lib/lists.nix

```diff
@@ -200,6 +200,46 @@ rec {
     let found = filter pred list;
     in if found == [] then default else head found;
 
+  /* Takes a predicate and a list as input and
+     returns a list of lists, separated by elements that match the predicate.
+
+     This is analoguous to `builtins.split separator string`,
+     with a predicate instead of a separator
+     and a list instead of a string.
+
+     Type: splitList :: (a -> bool) -> [a] -> [ (a | [a]) ]
+
+     The returned list always has the form `[ [...] sep [...] ... sep [...] ]`
+     i.e. begins with a list, alternates with separators and ends with a list,
+     so that the following is true for `res = splitList pred l`:
+     - `res` has length `2 * k + 1`,
+       where `k` is the number of elements of `l` that
+       match `pred`. (ie `splitList pred l`)
+     - `elemAt res (2 * n)` is always a list,
```

Comment:
Maybe add “where `n` is a number smaller than `builtins.div (builtins.length l) 2`. Same for the next property.